rank_bm25.py

```diff
@@ -14,45 +14,67 @@
 
 class BM25:
     def __init__(self, corpus, tokenizer=None):
-        self.corpus_size = 0
-        self.avgdl = 0
-        self.doc_freqs = []
-        self.idf = {}
-        self.doc_len = []
-        self.tokenizer = tokenizer
+        self.corpus_size = 0  # total number of document in corpus
+        self.avgdl = 0  # average length of a document in corpus
+        self.doc_freqs = []  # list of dictionaries of term_frequency of each document
+        self.idf = {}  # idf score of each word in whole corpus
+        self.doc_len = []  # list of length of each document in corpus
+        self.tokenizer = tokenizer  # user input tokenizer, defaults to none
 
         if tokenizer:
             corpus = self._tokenize_corpus(corpus)
 
         nd = self._initialize(corpus)
         self._calc_idf(nd)
 
-    def _initialize(self, corpus):
+    def _initialize(self, corpus) -> Dict:
+        """
+        Args:
+            corpus (List): list of the documents
+
+        Returns:
+            dict: number of document that contains the word
+
+        Example:
+            corpus = [['ram', 'is', 'a', 'good', 'boy'], ['ram', 'does', 'cycling', 'and', 'racing'], ['ram', 'is', 'healthy'], ['rita', 'likes', 'shyam'], ['good', 'luck']]
+            nd = {'ram': 3, 'is': 2, 'a': 1, 'good': 2, 'boy': 1, 'does': 1, 'cycling': 1, 'and': 1, 'racing': 1, 'healthy': 1, 'rita': 1, 'likes': 1, 'shyam': 1, 'luck': 1}
```

Comment:
Shorten the examples so that I don't need to scroll. The functionality can also be explained only using 2 items in the list.